stripe/src/main/java/com/stripe/android/EphemeralKeyManager.java

```diff
@@ -58,14 +58,29 @@ private void updateKey(
             @NonNull String key,
             @Nullable String actionString,
             @Nullable Map<String, Object> arguments) {
+        // Key is coming from the user, so even if it's @NonNull annotated we
+        // want to double check it
+        if (key == null) {
+            mListener.onKeyError(HttpURLConnection.HTTP_INTERNAL_ERROR,
+                    "EphemeralKeyUpdateListener.onKeyUpdate was called " +
+                            "with a null value");
+        }
         try {
             mEphemeralKey = AbstractEphemeralKey.fromString(key, mEphemeralKeyClass);
+            mListener.onKeyUpdate(mEphemeralKey, actionString, arguments);
         } catch (JSONException e) {
             mListener.onKeyError(HttpURLConnection.HTTP_INTERNAL_ERROR,
-                    "The JSON from the key could not be parsed: "
-                            + e.getLocalizedMessage());
+                    "EphemeralKeyUpdateListener.onKeyUpdate was passed " +
+                            "a value that could not be JSON parsed: ["
+                            + e.getLocalizedMessage() + "] the raw body from Stripe's response" +
```

Comment:
I think `"the raw body..."` should be a new sentence, so `"]. The raw body...`

Also below.

Reply:
Yes ok